hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1102,7 +1103,17 @@ def as_strategy(strat_or_callable, thing, final=True):
             f"Could not resolve {thing!r} to a strategy, because it is an abstract "
             "type without any subclasses. Consider using register_type_strategy"
         )
-    return sampled_from(subclasses).flatmap(from_type)
+    subclass_strategies = nothing()
+    for sc in subclasses:
+        try:
+            subclass_strategies |= _from_type(sc)
```

Comment:
What if we have other abstract subclasses? Like this:
- `class A(Abstract): ...`  is abstract
- `class B(A, Abstract): ...` is still abstract
- `class C(B): ...` is concrete

Reply:
A resolves to B|C, and B resolves to C, so we have a touch of redundancy in C|C but that's OK - given the whole register_subclass trick it's more complex than it's worth to avoid this.